source/FreeRTOS_IP_Utils.c

```diff
@@ -996,6 +996,10 @@ void vPreCheckConfigs( void )
                 }
             #endif
             /* LCOV_EXCL_BR_STOP */
+
+            /* ipIP_TYPE_OFFSET = ( sizeof( UDPHeader_t ) + sizeof( IPHeader_IPv6_t ) - ( sizeof( UDPPacket_t ) )
+             * and it MUST be > 0, otherwise storing the IPv4 version byte will overwrite the Ethernet header. */
+            configASSERT( ipIP_TYPE_OFFSET > 0 );
```

Comment:
As `ipIP_TYPE_OFFSET` is a summation of `sizeof()` expressions, the result is an *unsigned number*.
I just tested it with dummy structs and it did not assert when it should.